contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -419,6 +435,22 @@ transform_from_ci_as_for_likenode(Node *node, OpExpr *op, like_ilike_info_t like
 		return node;
 	}
 
+	/* 
+	 * Obtain the original typeId of leftop so that we can find the compatible =, >= and <
+	 * operator for the original typeId. Else we will always obtain the operators compatible
+	 * with TEXT datatype as the operands get type coerced into TEXT as LIKE is defined for it
+	 * Make the typeId of rightop same as leftop so that we obtain expected operator
+	 * Similarly, update the type of prefix to have appropriate datatypes of operands
+	 * 
+	 * Optimiser will remove Relabel Node during Index scan, see match_index_to_operand
+	 */
+	if (IsA(leftop, RelabelType))
+	{
+		RelabelType	*relabel = (RelabelType *) leftop;
+		leftop = copyObject((Node*) relabel->arg);
+		prefix->consttype = rtypeId = ltypeId = exprType(leftop);
+	}
```

Comment:
can we move this above where we are handing right operand for Relabel node?

Reply:
We can not because:
1. prefix->consttype is set here, howeverer prefix is initialised later
2. We only need to find the correct typeid if we want to have the optimisation, so logically this makes sense